backtester/funding/funding.go

```diff
@@ -564,30 +564,22 @@ func (f *FundManager) UpdateFundingFromLiveData(initialFundsSet bool) error {
 	if err != nil {
 		return err
 	}
-	for x := range exchanges {
-		var creds *account.Credentials
-		creds, err = exchanges[x].GetCredentials(context.TODO())
-		if err != nil {
-			return err
-		}
-		assets := exchanges[x].GetAssetTypes(false)
-		for y := range assets {
-			if assets[y].IsFutures() {
+	for _, e := range exchanges {
+		eName := e.GetName()
+		assets := e.GetAssetTypes(false)
+		for _, a := range assets {
+			if a.IsFutures() {
 				// we set all holdings as spot
 				// futures currency holdings are collateral in the collateral currency
 				continue
 			}
-			var acc account.Holdings
-			acc, err = exchanges[x].UpdateAccountInfo(context.TODO(), assets[y])
+			subAccts, err := e.UpdateAccountBalances(context.TODO(), a)
 			if err != nil {
 				return err
 			}
-			for z := range acc.Accounts {
-				if !acc.Accounts[z].Credentials.Equal(creds) {
-					continue
-				}
-				for i := range acc.Accounts[z].Currencies {
-					err = f.SetFunding(exchanges[x].GetName(), assets[y], &acc.Accounts[z].Currencies[i], initialFundsSet)
+			for _, subAcct := range subAccts {
+				for _, bal := range subAcct.Balances {
+					err = f.SetFunding(eName, a, &bal, initialFundsSet)
```

Comment:
can make it into a single like 
```suggestion
					if err = f.SetFunding(eName, a, &bal, initialFundsSet); err != nil {
```